### PR TITLE
Add ability to request manual rebuild of Delta CRLs

### PR DESCRIPTION
**Pull request by cipherboy**
Since Delta CRL functionality has its own changelog entry, I'm inclined not to bother with a changelog for this.

---

We add a new read endpoint, `crl/rotate-delta` to force a manual rebuild of the Delta CRLs on this local cluster. This mirrors the behavior of the existing `crl/rotate` endpoint, just for delta CRLs instead of full CRLs. We add a warning that this is a no-op when delta CRLs are not enabled.

This also adds a missing doc (as pointed out by @Gabrielopesantos in a ticket) about the response header. As far as I can tell, there's no real way to validate this header with `client.Logical()`; we'd probably have to resort to `client.RawRequest(...)` to validate it if we wanted to.

### CODE REVIEW

File: builtin/logical/pki/path_revoke.go [hunk 1]
```diff
@@ -414,6 +433,39 @@ func (b *backend) pathRotateCRLRead(ctx context.Context, req *logical.Request, _
 	}, nil
 }
 
+func (b *backend) pathRotateDeltaCRLRead(ctx context.Context, req *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	sc := b.makeStorageContext(ctx, req.Storage)
```
Comment (cipherboy) [inline builtin/logical/pki/path_revoke.go:437]: I'm curious what the thoughts are here about locks. The builder itself is locked, but `pathRotateCRLRead` grabs the storage lock, which I think feels wrong.

File: builtin/logical/pki/path_revoke.go [hunk 2]
```diff
@@ -104,6 +104,25 @@ func pathRotateCRL(b *backend) *framework.Path {
 	}
 }
 
+func pathRotateDeltaCRL(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `crl/rotate-delta`,
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
```
Comment (stevendpclark) [inline builtin/logical/pki/path_revoke.go:112]: *sigh* I wish this were a POST but I suppose it makes more sense to follow the crl/rotate way of things

**Review by stevendpclark — APPROVED** [review]

File: builtin/logical/pki/path_revoke.go (hunk 2 above)
Comment (cipherboy) [reply builtin/logical/pki/path_revoke.go:112]: We could add both as Update operations if we wanted I suppose?

**Comment by cipherboy** [comment]
Thanks! Auto-merging...

File: builtin/logical/pki/path_revoke.go (hunk 1 above)
Comment (stevendpclark) [reply builtin/logical/pki/path_revoke.go:437]: I'm in agreement, if anything should be grabbing a lock it should be within the crlBuilder's code. Probably something I missed while refactoring?

**Pull request merged**